### DISCUSSION TITLE
disk tags not working

### DISCUSSION THREAD

**Discussion by ypyly** [longhorn/longhorn#3478] [Q&A] [answered]
hi, I've faced an issue with working Longhorn installation when I've tried to add additional HDD's to the system
I've tagged nodes and disks, created appropriate storage class in kubernetes, but whenever I'm trying to create pvc/pv, it still utilizes default disk.
If I'm creating volume from the Longhorn UI - everything works as expected, I can validate, that physical location of the volume is correct and I can mount it to the node without any issue.
I suppose, that storage classes and tags (via UI) are defined properly, what should I check additionally? 

Node disks and tags
![2022-01-01 14_37_00-Longhorn](https://user-images.githubusercontent.com/50587039/147850708-44f04ca1-54eb-49be-8f91-3b064854a1b0.png)
Creation of volume manually: 
![2022-01-01 14_38_33-creation](https://user-images.githubusercontent.com/50587039/147850801-01dbfaf7-e24f-4cf7-9851-e2417d389c16.png)
created with proper path:
![2022-01-01 14_39_04-created](https://user-images.githubusercontent.com/50587039/147850808-71262f61-f5e0-4413-8d21-eb13335cdb93.png)
and attached:
![2022-01-01 14_39_44-attached](https://user-images.githubusercontent.com/50587039/147850810-ce3f6a7f-9e6b-4ce4-8131-c0d357dbb42a.png)

Storage Classes:
`
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: longhorn-fast
provisioner: driver.longhorn.io
allowVolumeExpansion: true
parameters:
  numberOfReplicas: "4"
  staleReplicaTimeout: "30"
  diskSelector: "fast"
  nodeSelector: "storage"
  fromBackup: ""
  fstype: "xfs"
`
`
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: longhorn-slow
provisioner: driver.longhorn.io
allowVolumeExpansion: true
parameters:
  numberOfReplicas: "4"
  staleReplicaTimeout: "30"
  discSelector: "slow"
  nodeSelector: "storage"
  fromBackup: ""
  fstype: "xfs"`

Deployment:
`apiVersion: apps/v1
kind: Deployment
metadata:
  name: test
  labels:
    app: test
spec:
  replicas: 1
  strategy: 
    type: Recreate
  selector:
    matchLabels:
      app: test
  template:
    metadata:
      labels:
        app: test
    spec:
      containers:
      - name: test
        resources:
          requests:
            cpu: "1"
            memory: "2Gi"
          limits:
            cpu: "2"
            memory: "4Gi"
        image: ubuntu
        name: ubuntu
        command: ["/bin/sleep", "3650d"]
        volumeMounts:
         - mountPath: "/mnt/data"
           name: test
           readOnly: false
      volumes:
        - name: test
          persistentVolumeClaim:
            claimName: pv-test
`
PVC: 
`
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pv-test
spec:
  storageClassName: longhorn-slow
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
`
immediately after apply:
![2022-01-01 14_46_31-apply](https://user-images.githubusercontent.com/50587039/147850948-5a8f27eb-2003-40e3-b33f-a32862acc9fe.png)

couple of seconds later:
![2022-01-01 14_49_23-after apply](https://user-images.githubusercontent.com/50587039/147850910-458b1d36-4f40-4bcb-aa82-93a3ebe4e18b.png)
while manually created volume, persists in proper location
What could be the problem?

**Comment by jenting** [comment, marked as answer]
I saw there is a typo in your manifest `discSelector`, should be `diskSelector`.
Is this the root cause?

**Reply by ypyly** [reply]
I kindly thank you for your input - I've been bashing my head against the wall for couple of days already (during my free time) and was not focused enough to see a dumb typo. THANK YOU
